library/EasyPipeline/src/Easy.hs

```diff
@@ -36,59 +39,77 @@ import qualified Juvix.Frontend.Types.Base as Frontend
 import qualified Juvix.FrontendDesugar as FrontDesugar
 import Juvix.Library
 import qualified Juvix.Library.Feedback as Feedback
+import qualified Juvix.Library.HashMap as Map
 import qualified Juvix.Library.NameSymbol as NameSymb
 import qualified Juvix.Library.Sexp as Sexp
 import qualified Juvix.Pipeline as Pipeline
 import qualified Juvix.Pipeline.Compile as Compile
+import qualified Juvix.ToCore.Types as ToCore.Types
 import qualified Text.Pretty.Simple as Pretty
 import Prelude (error)
+import qualified Prelude (Show (..))
 
 --------------------------------------------------------------------------------
 -- OPTIONS
 --------------------------------------------------------------------------------
 
-data Options = Opt
+instance Prelude.Show (Core.Parameterisation primTy primVal) where
+  show _ = "param"
+
+data Options primTy primVal = Opt
   { prelude :: [FilePath],
-    currentContextName :: NameSymb.T
+    currentContextName :: NameSymb.T,
+    param :: Core.Parameterisation primTy primVal
   }
   deriving (Show)
 
 -- we can override defaults by saying def { newOptions }
-def :: Options
+def :: Options primTy primVal
 def =
   Opt
     { -- to avoid being overhwlemed in the repl by giant text, we have
       -- a minimal file here. Our functions will take def, so we can
       -- replace it by the full library
       prelude = ["juvix/minimal.ju"],
       -- by default our code will live in Juvix-User
-      currentContextName = "Juvix-User"
+      currentContextName = "Juvix-User",
+      param = undefined
     }
 
 -- @defMichelson@ gives us Michelson prelude
-defMichelson :: Options
+defMichelson :: Options Michelson.Param.PrimTy Michelson.Param.RawPrimVal
 defMichelson =
   def
     { prelude =
         -- TODO: Avoid relative paths
         [ "../../stdlib/Prelude.ju",
           "../../stdlib/Michelson.ju",
           "../../stdlib/MichelsonAlias.ju"
-        ]
+        ],
+      param = Michelson.Param.michelson
     }
 
 -- @defCircuit@ gives us the circuit prelude
-defCircuit :: Options
+-- defCircuit :: Options
```

Comment:
Why no type specification?

Reply:
Requires another library to be imported IIRC, so I just left this one out